src/shillelagh/adapters/api/github.py

```diff
@@ -58,45 +59,110 @@ class Column:
-        ],
+        "pulls": {
+            "columns": [
+                Column("url", "html_url", String()),
+                Column("id", "id", Integer()),
+                Column("number", "number", Integer(filters=[Equal])),
+                Column("state", "state", String(filters=[Equal]), Equal("all")),
+                Column("title", "title", String()),
+                Column("userid", "user.id", Integer()),
+                Column("username", "user.login", String()),
+                Column("draft", "draft", Boolean()),
+                Column("head", "head.ref", String(filters=[Equal])),  # head.label?
+                Column("created_at", "created_at", StringDateTime()),
+                Column("updated_at", "updated_at", StringDateTime()),
+                Column("closed_at", "closed_at", StringDateTime()),
+                Column("merged_at", "merged_at", StringDateTime()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
+        "issues": {
+            "columns": [
+                Column("url", "html_url", String()),
+                Column("id", "id", Integer()),
+                Column("number", "number", Integer(filters=[Equal])),
+                Column("state", "state", String(filters=[Equal]), Equal("all")),
+                Column("title", "title", String()),
+                Column("userid", "user.id", Integer()),
+                Column("username", "user.login", String()),
+                Column("draft", "draft", Boolean()),
+                Column("locked", "locked", Boolean()),
+                Column("comments", "comments", Integer()),
+                Column("created_at", "created_at", StringDateTime()),
+                Column("updated_at", "updated_at", StringDateTime()),
+                Column("closed_at", "closed_at", StringDateTime()),
+                Column("body", "body", String()),
+                Column("author_association", "author_association", String()),
+                Column("labels", "labels[*].name", JSONString()),
+                Column("assignees", "assignees[*].login", JSONString()),
+                Column("reactions", "reactions", JSONString()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
+        "stats/punch_card": {
+            "columns": [
+                Column("dow", "$.[0]", Integer()),
+                Column("hour", "$.[1]", Integer()),
+                Column("commits", "$.[2]", Integer()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
+        "stats/participation": {
+            "columns": [
+                Column("start_at", "$.start_at", DateTime()),
+                Column("end_at", "$.end_at", DateTime()),
+                Column("all", "$.all", Integer()),
+                Column("owner", "$.owner", Integer()),
+            ],
+            "paginated": False,
+            "processor": participation_processor,
+        },
```

Comment:
@rusackas this one was a bit more complicated, since the payload looks like this:

```json
{
  "all": [27, 34, 27, 31, 24, ...],
  "owner": [0, 0, 0, 0, 0, ...],
}
```

Here, the documentation says:

> The array order is oldest week (index 0) to most recent week. The most recent week is seven days ago at UTC midnight to today at UTC midnight.

So I wrote a custom processor that actually computes the bounding timestamps and adds them to the new columns `start_at` and `end_at`.

Reply:
Also, what the hell does "today midnight" mean? Is it the midnight of the beginning of the day or at the end? I assume it's at the beginning, otherwise the numbers would change throughout the day.